File: fixtures/scale/scala/large/src/cryptobench/ciphers/rot13/Rot13Suite.scala
package cryptobench.ciphers.rot13

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class Rot13Suite extends CipherSuite {
  override def name: String = "rot13"

  override def run(): SuiteResult = {
    val cipher: Cipher = new Rot13Cipher(Rot13Key.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- Rot13Vectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}

File: fixtures/scale/dart/large/lib/ciphers/rot13/rot13_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'rot13_cipher.dart';
import 'rot13_key.dart';
import 'rot13_vectors.dart';

class Rot13Suite implements CipherSuite {
  @override
  String name() => 'rot13';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = Rot13Cipher(Rot13Key.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in rot13Samples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
